src/coreclr/jit/lsra.h

```diff
@@ -1595,9 +1605,12 @@ class LinearScan : public LinearScanInterface
     // A temporary VarToRegMap used during the resolution of critical edges.
     VarToRegMap sharedCriticalVarToRegMap;
 
-    PhasedVar<regMaskTP>  availableIntRegs;
-    PhasedVar<regMaskTP>  availableFloatRegs;
-    PhasedVar<regMaskTP>  availableDoubleRegs;
+    PhasedVar<regMaskTP> availableIntRegs;
+    PhasedVar<regMaskTP> availableFloatRegs;
+    PhasedVar<regMaskTP> availableDoubleRegs;
```

Comment:
@kunalspathak, it's not clear why we have `availableFloatRegs` and `availableDoubleRegs`. Could you elaborate a bit?

Given that `availableDoubleRegs` is used to account for both `double` and `TYP_SIMD` and that `float`/`double`/`simd` are all the "same registers", it's not immediately clear why we have the split.

Even on Arm32 where you have 1x double register taking up 2x float registers, it seems like it'd be simpler/easier to track this as a single "register file" and for double to simply take up 2 bits.

Reply:
-- Not something to handle in this PR, more making observations as this code is being touched in general and looking for potential opportunities to reduce the impact of changes like this, particularly given how impactful LSRA changes are to overall TP.

Reply:
Honestly, I don't know historical reason behind having 2 separate variables for floats and double although we have `#define RBM_ALLDOUBLE            RBM_ALLFLOAT` in all targets except ARM. I will have to spend some time in understanding the implication if we unify them.